### PR TITLE
Add documentations for Installation

### PR DESCRIPTION
**Pull request by zhaochenyang20**
## Motivation

I added documents for the installation right now. Other files like `docs/en/quick_start.md` could be ignored and I will wrap it into `index.rst` later.

## Modifications

https://github.com/sgl-project/sglang/issues/1487

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: docs/en/install.md [hunk 1]
```diff
@@ -1,56 +1,75 @@
...
-### Method 2: From source
-```
-# Use the last release branch
-git clone -b v0.3.0 https://github.com/sgl-project/sglang.git
-cd sglang
+**Note: The backend (SRT) is only needed on the server side and is only available for Linux right now.**
 
-pip install --upgrade pip
-pip install -e "python[all]"
+**Important: Please check the [flashinfer installation guidance](https://docs.flashinfer.ai/installation.html) to install the proper version according to your PyTorch and CUDA versions.**
```
Comment (merrymercy) [inline docs/en/install.md]: useful

File: docs/en/install.md [hunk 2]
```diff
@@ -1,56 +1,75 @@
-## Install SGLang
+# Installation Guide
 
-You can install SGLang using any of the methods below.
+SGLang comprises a frontend language, namely the Structured Generation Language (SGLang), and a backend runtime known as the SGLang Runtime (SRT).
 
-### Method 1: With pip
+The frontend can be utilized independently of the backend. This implies that you can install the frontend language on a local device without a GPU and install the backend runtime on a GPU cluster. As a result, you can construct your LM programs locally and establish a connection with the remote backend runtime.
```
Comment (merrymercy) [inline docs/en/install.md]: useful, move it to #Common Notes

File: docs/en/install.md [hunk 3]
```diff
@@ -1,56 +1,75 @@
...
-pip install --upgrade pip
-pip install -e "python[all]"
+**Important: Please check the [flashinfer installation guidance](https://docs.flashinfer.ai/installation.html) to install the proper version according to your PyTorch and CUDA versions.**
 
-# Install FlashInfer CUDA kernels
+### 3. From Source (Latest version, Linux only for full installation)
+
+```bash
+# Use the latest release branch
+# As of this documentation, it's v0.3.4, but newer versions may be available
```
Comment (merrymercy) [inline docs/en/install.md]: make it more concise.

File: docs/en/install.md [hunk 4]
```diff
@@ -1,56 +1,75 @@
...
 
 ```bash
-docker run --gpus all \
-    -p 30000:30000 \
+pip install "sglang[openai]"
+```
+
+## Advanced Installation Options
+
+### 1. Using Docker (Server-side, Linux only)
```
Comment (merrymercy) [inline docs/en/install.md]: this is not "advanced"

File: docs/en/install.md [hunk 5]
```diff
@@ -91,6 +110,9 @@ sky status --endpoint 30000 sglang
 3. To further scale up your deployment with autoscaling and failure recovery, check out the [SkyServe + SGLang guide](https://github.com/skypilot-org/skypilot/tree/master/llm/sglang#serving-llama-2-with-sglang-for-more-traffic-using-skyserve).
 </details>
 
-### Common Notes
-- [FlashInfer](https://github.com/flashinfer-ai/flashinfer) is the default attention kernel backend. It only supports sm75 and above. If you encounter any FlashInfer-related issues on sm75+ devices (e.g., T4, A10, A100, L4, L40S, H100), please switch to other kernels by adding `--attention-backend triton --sampling-backend pytorch` and open an issue on GitHub.
-- If you only need to use the OpenAI backend, you can avoid installing other dependencies by using `pip install "sglang[openai]"`.
+## Troubleshooting
```
Comment (merrymercy) [inline docs/en/install.md]: fix this

File: docs/en/quick_start.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
+# Server Quick Start
+
+This guide will help you quickly set up and run the SGLang server, and send test requests for both generative and embedding models.
+
```
Comment (merrymercy) [inline docs/en/quick_start.md]: Current outline
```
- Step 1
   - Generative model (90% user)
   - Embedding model  (10% user)
- Step 2
   - Generative model (90% user)
   - Embedding model  (10% user)
```
   
Correct outline
```
- Generative model (90% user)
  - Step 1
  - Step 2
- Embedding model (10% user)
  - Step 1
  - Step 2
```

**Comment by merrymercy** [comment]
Principal:
- Common case first, move edge cases to the end
- Make it concise
- Make the commands runnable and continuous

**Comment by zhaochenyang20** [comment]
@merrymercy Hey, plz check if this version of `install.md` is what you'd prefer.

File: docs/en/backend_overview.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# Backend Overview
```
Comment (ByronHsu) [inline docs/en/backend_overview.md]: Is this file necessary? i think the paper already provides a good overview

File: docs/en/index.rst [hunk 8]
```diff
@@ -1,14 +1,20 @@
 SGLang Documentation
 ====================================
 
-SGLang is a fast serving framework for large language models and vision language models.
-It makes your interaction with models faster and more controllable by co-designing the backend runtime and frontend language.
-The core features include:
+Welcome to SGLang!
```
Comment (ByronHsu) [inline docs/en/index.rst]: This seems redundant

File: docs/en/backend_overview.md (hunk 7 above)
Comment (zhaochenyang20) [reply docs/en/backend_overview.md]: This document serves as a learning guide, and I believe it is necessary. As you can gain insights from the Stoicheia, there remains a need to acquire new Geometry material. Anyway, I will set this aside for now and ensure it is placed appropriately later.

File: docs/en/index.rst (hunk 8 above)
Comment (zhaochenyang20) [reply docs/en/index.rst]: I don't think it's redundant. I regard it's appropriate.

Welcome to SGLang!

SGLang is a fast, open-source framework for serving large and vision-language models. It enhances speed and control by co-designing the backend runtime and frontend language. Join our vibrant community and contribute to advancing LLM and VLM serving.

The core features of SGLang include:

Fast Backend Runtime: Efficient serving with RadixAttention for prefix caching, jump-forward constrained decoding, continuous batching, token attention (paged attention), tensor parallelism, flashinfer kernels, and quantization (AWQ/FP8/GPTQ/Marlin).

Flexible Frontend Language: Enables easy programming of LLM applications with chained generation calls, advanced prompting, control flow, multiple modalities, parallelism, and external interactions.

Extensive Model Support: SGLang supports a wide range of generative models including the Llama series (up to Llama 3.1), Mistral, Gemma, Qwen, DeepSeek, LLaVA, Yi-VL, StableLM, Command-R, DBRX, Grok, ChatGLM, InternLM 2 and Exaone 3. It also supports embedding models such as e5-mistral and gte-Qwen2. Easily extensible to support new models.

File: docs/en/add_new_model.md
Comment (merrymercy) [inline docs/en/add_new_model.md:1]: revert this change.

File: docs/en/index.rst [hunk 9]
```diff
@@ -1,14 +1,20 @@
 SGLang Documentation
```
Comment (merrymercy) [inline docs/en/index.rst:1]: revert the change of this file, and only keep `install.md`

File: docs/en/add_new_model.md [hunk 10]
```diff
@@ -1,4 +1,4 @@
-# How to Support a New Model
+# Add New Models
```
Comment (merrymercy) [inline docs/en/add_new_model.md]: revert the change of this file, and only keep `install.md`

File: docs/en/index.rst (hunk 9 above)
Comment (merrymercy) [inline docs/en/index.rst]: revert the change of this file, and only keep `install.md`

File: docs/en/install.md [hunk 11]
```diff
@@ -1,41 +1,41 @@
...
+# Installation Guide
 
 ### Method 1: With pip
-```
+
+```bash
 pip install --upgrade pip
 pip install "sglang[all]"
-
-# Install FlashInfer CUDA kernels
```
Comment (merrymercy) [inline docs/en/install.md]: keep this

File: docs/en/install.md [hunk 12]
```diff
@@ -1,41 +1,41 @@
...
-docker run --gpus all \
-    -p 30000:30000 \
+docker run --gpus all -p 30000:30000 \
     -v ~/.cache/huggingface:/root/.cache/huggingface \
-    --env "HF_TOKEN=<secret>" \
-    --ipc=host \
+    --env "HF_TOKEN=<secret>" --ipc=host \
     lmsysorg/sglang:latest \
-    python3 -m sglang.launch_server --model-path meta-llama/Llama-3.1-8B-Instruct --host 0.0.0.0 --port 30000
+    python3 -m sglang.launch_server --model-path meta-llama/Meta-Llama-3.1-8B-Instruct --host 0.0.0.0 --port 30000
```
Comment (merrymercy) [inline docs/en/install.md]: revert this line

**Review by merrymercy — CHANGES_REQUESTED** [review]

**Comment by zhaochenyang20** [comment]
thanks for help @merrymercy @ByronHsu
